source/Zyan.Communication/Security/BasicWindowsAuthProvider.cs

```diff
@@ -79,18 +79,14 @@ public AuthResponseMessage Authenticate(AuthRequestMessage authRequest)
 
 			// Benutzer und Kennwort lesen
 			string userName = authRequest.Credentials[AuthRequestMessage.CREDENTIAL_USERNAME] as string;
-			string password = authRequest.Credentials[AuthRequestMessage.CREDENTIAL_PASSWORD] as string;
+			userName = GetWorkingUserName(userName);
+			string userPassword = authRequest.Credentials[AuthRequestMessage.CREDENTIAL_PASSWORD] as string;
 
 			// Variable für Domäne
-			string domain = ".";
-
-			// Wenn eine Domäne angegeben wurde ...
-			if (authRequest.Credentials.ContainsKey(AuthRequestMessage.CREDENTIAL_DOMAIN))

```

Comment:
Can we still have this logic? Something like

```csharp
var domain = GetJoinedDomainName();

// if a user provides his domain name, use it
if (authRequest.Credentials.ContainsKey(AuthRequestMessage.CREDENTIAL_DOMAIN))
{
   domain = authRequest.Credentials[AuthRequestMessage.CREDENTIAL_DOMAIN] as string;
}
```

Reply:
Hi @yallie ,

my thoughts were that it does not matter, which domain name is in the credentials because, the server can only check against the own domain.

But i agree with you, that because of backward compatibility the check is useful ;)

Reply:
Yes, that would be just fine, too :)